container/src/main/java/com/flipkart/poseidon/Poseidon.java

```diff
@@ -149,14 +149,17 @@ private Handler getRewriteHandler(Handler handler) {
         rewriteHandler.setHandler(handler);
 
         try {
-            JavaType listRuleType = getMapper().getTypeFactory().constructParametricType(List.class, RewriteRule.class);
-            List<RewriteRule> rules = getMapper().readValue(new FileInputStream(configuration.getRewriteFilePath()), listRuleType);
-            for (RewriteRule rule : rules) {
-                if (rule.isActive()) {
-                    RewriteRegexRule regexRule = new RewriteRegexRule();
-                    regexRule.setRegex(rule.getFrom());
-                    regexRule.setReplacement(rule.getTo());
-                    rewriteHandler.addRule(regexRule);
+            String rewriteFilePath = configuration.getRewriteFilePath();
+            if (rewriteFilePath != null && !(rewriteFilePath = rewriteFilePath.trim()).isEmpty()) {
+                JavaType listRuleType = getMapper().getTypeFactory().constructParametricType(List.class, RewriteRule.class);
+                List<RewriteRule> rules = getMapper().readValue(new FileInputStream(rewriteFilePath), listRuleType);
+                for (RewriteRule rule : rules) {
+                    if (rule.isActive()) {
+                        RewriteRegexRule regexRule = new RewriteRegexRule();
+                        regexRule.setRegex(rule.getFrom());
+                        regexRule.setReplacement(rule.getTo());
+                        rewriteHandler.addRule(regexRule);
+                    }
```

Comment:
Shrey, can you test this before I merge this PR? I mean from your app don't supply rewriteFilePath and make sure APIs work without any exceptions (mainly because we are still creating RewriteHandler). Another question is do we need to create a rewriteHandler when no rewrite file path isn't specified?